public/app/features/plugins/extensions/registry/types.ts

```diff
@@ -0,0 +1,15 @@
+import { AddedComponentRegistryItem, AddedComponentsRegistry } from './AddedComponentsRegistry';
+import { AddedLinkRegistryItem, AddedLinksRegistry } from './AddedLinksRegistry';
+import { ExposedComponentsRegistry } from './ExposedComponentsRegistry';
+import { RegistryType } from './Registry';
+
+export type PluginRegistryStates = {
```

Comment:
Hmm, I find this a little bit misleading (or the name), as it's missing the registry for the exposed components, and although I understand why, I think it can be a bit confusing when compared to the next type (`PluginExtensionRegistries`). As it's right now only used in one place I would probably just inline it there. What do you think?

```ts
let registryStates: {
  addedComponentsRegistry: RegistryType<AddedComponentRegistryItem[]>;
  addedLinksRegistry: RegistryType<AddedLinkRegistryItem[]>;
} = 
```

Reply:
Yep totally agree. Changed to use inline types 👍